packages/react/select/src/Select.tsx

```diff
@@ -243,18 +245,20 @@ const VALUE_NAME = 'SelectValue';
 
 type SelectValueElement = React.ElementRef<typeof Primitive.span>;
 type PrimitiveSpanProps = Radix.ComponentPropsWithoutRef<typeof Primitive.span>;
-interface SelectValueProps extends PrimitiveSpanProps {}
+interface SelectValueProps extends Omit<PrimitiveSpanProps, 'placeholder'> {
+  placeholder?: React.ReactNode;
+}
 
 const SelectValue = React.forwardRef<SelectValueElement, SelectValueProps>(
   (props: ScopedProps<SelectValueProps>, forwardedRef) => {
     // We ignore `className` and `style` as this part shouldn't be styled.
-    const { __scopeSelect, className, style, ...valueProps } = props;
+    const { __scopeSelect, className, style, children, placeholder, ...valueProps } = props;
     const context = useSelectContext(VALUE_NAME, __scopeSelect);
     const { onValueNodeHasChildrenChange } = context;
-    const hasChildren = props.children !== undefined;
+    const hasChildren = children !== undefined;
```

Comment:
One of the difficult things for me to figure out was that `value` has to be `undefined` for the `placeholder` to be shown. I used to use `null` or `''` for empty (reset) value, and not just missing. I understand it's a wrong habit to have, just want others to be clear about it.

Reply:
This is a veeeery good point, and I just spent 1 hour debugging why my placeholder was not showing up...